app/Http/Controllers/Api/ChannelEpisodesController.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Suyabay\Http\Controllers\Api;
+
+use DB;
+use Validator;
+use Suyabay\Channel;
+use Suyabay\Episode;
+use Suyabay\Http\Requests;
+use League\Fractal\Manager;
+use Illuminate\Http\Request;
+use League\Fractal\Resource\Item;
+use Illuminate\Mail\Mailer as Mail;
+use League\Fractal\Resource\Collection;
+use Illuminate\Support\Facades\Response;
+use Suyabay\Http\Controllers\Controller;
+use Suyabay\Http\Repository\ChannelRepository;
+use Suyabay\Http\Transformers\ChannelEpisodesTransformer;
+
+class ChannelEpisodesController extends Controller
+{
+    protected $mail;
+    protected $fractal;
+
+    public function __construct(Manager $fractal, Mail $mail)
+    {
+        $this->fractal = $fractal;
+
+        parent::__construct($mail);
+    }
+
+    /**
+     * This method takes channel name and return all episodes podcast under it.
+     *
+     * @param $request
+     * @param $name
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function getAllEpisodes(Request $request, $name, ChannelEpisodesTransformer $channelEpisodesTransformer)
```

Comment:
**Code Quality - Nitpick**
@andela-tolotin Please place the parameter coming from the URL before all other parameters for this method